com.oracle.truffle.r.native/fficall/src/truffle_common/altrep.c

```diff
@@ -0,0 +1,273 @@
+
+// TODO: Tady musim neco includovat
+#include <Rinternals.h>
+#include <R_ext/Altrep.h>
+#include "rffi_upcalls.h"
+
+int (ALTREP)(SEXP x) {
+    return ((call_ALTREP) callbacks[ALTREP_x])(x);
+}
+
+Rboolean R_altrep_inherits(SEXP instance, R_altrep_class_t class_descriptor) {
+    return ((call_R_altrep_inherits) callbacks[R_altrep_inherits_x])(instance, class_descriptor);
+}
+
+SEXP R_altrep_data1(SEXP x) {
+    return ((call_R_altrep_data1) callbacks[R_altrep_data1_x])(x);
+}
+
+SEXP R_altrep_data2(SEXP x) {
+    return ((call_R_altrep_data2) callbacks[R_altrep_data2_x])(x);
+}
+
+void R_set_altrep_data1(SEXP instance, SEXP data1) {
+    ((call_R_set_altrep_data1) callbacks[R_set_altrep_data1_x])(instance, data1);
+}
+
+void R_set_altrep_data2(SEXP instance, SEXP data2) {
+    ((call_R_set_altrep_data2) callbacks[R_set_altrep_data2_x])(instance, data2);
+}
+
+/**** Altrep method setters ****/
```

Comment:
Tyhle vsechny setter metody by se mozna daly vyresit nejakym makrem - podobne jako to ma GNU-R v `Altrep.h`